clang/lib/AST/ASTContext.cpp

```diff
@@ -831,6 +831,28 @@ ASTContext::getCanonicalTemplateTemplateParmDecl(
   return CanonTTP;
 }
 
+/// Check if a type can have its sanitizer instrumentation elided.
+/// Determine this by its presence in a SCL alongside its specified categories.
+/// For example:
+/// ignorelist.txt>
+/// [{unsigned-integer-overflow,signed-integer-overflow}]
+/// type:*=no_sanitize
+/// type:size_t=sanitize
+/// <ignorelist.txt
+/// Supplying the above ignorelist.txt will disable overflow sanitizer
+/// instrumentation for all types except "size_t".
+bool ASTContext::isTypeIgnoredBySanitizer(const SanitizerMask &Mask,
+                                          const QualType &Ty) const {
+  bool sanitizeType =
```

Comment:
Each contains call is loop over regexp list, relatively expensive.

Taking that into account, please:
1.  use lazy eval of boolean
2.  remove no_sanitize?
```
return NoSanitizeL->containsType(Mask, Ty.getAsString()) ||
         !NoSanitizeL->containsType(Mask, Ty.getAsString(), "sanitize");
```

Then the first call `NoSanitizeL->containsType` will be enough in most cases.

Reply:
We can't exactly drop the `no_sanitize` entirely as the `containsType()` check without a category specified will not match against types with the category `no_sanitize`. So if we removed the check from `isTypeIgnoredBySanitizer()` we would also have to remove `=no_sanitize` feature all together and just implicitly have this information from the lack of a category.

In short, let's keep the `no_sanitize` check and perhaps I could add some sort of caching system for these type checks? Keep in mind these only run when sanitizers are enabled and after a bunch of other checks too. The runtime hit should be minimal

Reply:
With that being said, I did update the logic a bit to avoid useless checks: https://github.com/llvm/llvm-project/pull/107332/commits/651bc7d2872a8f2b74a7c940f0ebe262e31dfc90

Reply:
> remove =no_sanitize feature all together

yes, that what I am asking, maybe we can drop it all together?
Probably performance diff will be hard to measure.

But ignoring thef perf, maybe I am missing something, but I don't see why we need 'no_sanitize' if there is `default`no category  approach, and it's already commonly used.

nit:  we should store `getAsString()` from the first call and reuse.

Reply:
I think `no_sanitize` will make it easier for users to understand what is happening with the ignorelists. Of course we will always support the default of no category. The performance difference is negligible: we will go from the best case of 1 call to `isTypeIgnoredBySanitizer()` up to 2 calls to that method.

Reply:
> I think no_sanitize will make it easier for users to understand

if we omit `no_sanitize` now we can add easily in future, if this is an issue. Removing `no_sanitize` will be harder.

So if we can avoid new category, I propose to avoid.

Reply:
hi, I removed `no_sanitize` with https://github.com/llvm/llvm-project/pull/107332/commits/b4a9448d2c8478849fb38d7066f33a609b76fe7b

thanks